### PR TITLE
Use SFx.NET to instrument cartservice

### PR DESCRIPTION
**Pull request by pjanotti**
## Why

Use the same instrumentation as currently recommended to customers.

## What

1. Add the SFx.NET auto-instrumentation to the cartservice docker image
2. Change the manual instrumentation on cartservice to use OpenTracing since this is the recommended for manual instrumentation when using SFx.NET
3. Enabled runtime metrics and thread sampling (new features)
4. Fixed the cartservice.csproj so we can build outside the docker image via dotnet CLI, also updated target framework, and NuGet packages.
5. Added a docker-compose to be used in tests
6. Removed Windows only scripts used for tests since now we can use the docker-compose

## Tests

Sample trace generated using docker-compose on my machine: https://app.signalfx.com/#/apm/traces/57ceb4d6b6d9cd4d3a050746b6b46b6f

## For the Future

Change to a more sophisticated application and

### CODE REVIEW

File: kubernetes-manifests/cartservice.yaml [hunk 1]
```diff
@@ -45,14 +45,17 @@ spec:
             fieldRef:
               fieldPath: status.hostIP
         - name: SIGNALFX_ENDPOINT_URL 
-          # value: "http://zipkin.default:9411/api/v2/spans"
           value: "http://$(NODE_IP):9411/api/v2/spans"
-        - name: SIGNALFX_SERVICE_NAME
-          value: "cartservice"
-        - name: SIGNALFX_SERVER_TIMING_CONTEXT
+        - name: SIGNALFX_LOGS_ENDPOINT_URL
+          value: "http://$(NODE_IP):4318/v1/logs"
```
Comment (dmitryax) [inline kubernetes-manifests/cartservice.yaml:50]: What king of logs are being sent from by the instrumentation?

Comment (pjanotti) [reply kubernetes-manifests/cartservice.yaml:50]: Profiling

Comment (dmitryax) [reply kubernetes-manifests/cartservice.yaml:50]: Just curious, is [this library](https://github.com/signalfx/signalfx-dotnet-tracing) sending profiling data in OTLP logs?

**Review by dmitryax — APPROVED** [review]

File: kubernetes-manifests/cartservice.yaml (hunk 1 above)
Comment (pjanotti) [reply kubernetes-manifests/cartservice.yaml:50]: correct

Comment (dmitryax) [reply kubernetes-manifests/cartservice.yaml:50]: Ok, thanks!

**Pull request merged**